streams/src/test/java/org/apache/kafka/streams/tests/EosTestDriver.java

```diff
@@ -68,72 +69,115 @@ private static synchronized void updateNumRecordsProduces(final int delta) {
+            final Map<Integer, List<Long>> offsets = new HashMap<>();
 
-            final ProducerRecord<String, Integer> record = new ProducerRecord<>("data", key, value);
+            while (isRunning) {
+                final String key = "" + rand.nextInt(MAX_NUMBER_OF_KEYS);
+                final int value = rand.nextInt(10000);
 
-            producer.send(record, (metadata, exception) -> {
-                if (exception != null) {
-                    exception.printStackTrace(System.err);
-                    System.err.flush();
-                    if (exception instanceof TimeoutException) {
-                        try {
-                            // message == org.apache.kafka.common.errors.TimeoutException: Expiring 4 record(s) for data-0: 30004 ms has passed since last attempt plus backoff time
-                            final int expired = Integer.parseInt(exception.getMessage().split(" ")[2]);
-                            updateNumRecordsProduces(-expired);
-                        } catch (final Exception ignore) { }
+                final ProducerRecord<String, Integer> record = new ProducerRecord<>("data", key, value);
+
+                producer.send(record, (metadata, exception) -> {
+                    if (exception != null) {
+                        exception.printStackTrace(System.err);
+                        System.err.flush();
+                        if (exception instanceof TimeoutException) {
+                            try {
+                                // message == org.apache.kafka.common.errors.TimeoutException: Expiring 4 record(s) for data-0: 30004 ms has passed since last attempt plus backoff time
+                                final int expired = Integer.parseInt(exception.getMessage().split(" ")[2]);
+                                updateNumRecordsProduces(-expired);
+                            } catch (final Exception ignore) {
+                            }
+                        }
+                    } else {
+                        offsets.getOrDefault(metadata.partition(), new LinkedList<>()).add(metadata.offset());
                     }
+                });
+
+                updateNumRecordsProduces(1);
+                if (numRecordsProduced % 1000 == 0) {
+                    System.out.println(numRecordsProduced + " records produced");
+                    System.out.flush();
                 }
-            });
+                Utils.sleep(rand.nextInt(10));
+            }
+            producer.close();
+            System.out.println("Producer closed: " + numRecordsProduced + " records produced");
+            System.out.flush();
 
-            updateNumRecordsProduces(1);
-            if (numRecordsProduced % 1000 == 0) {
-                System.out.println(numRecordsProduced + " records produced");
-                System.out.flush();
+            // verify offsets
+            for (final Map.Entry<Integer, List<Long>> offsetsOfPartition : offsets.entrySet()) {
```

Comment:
Does this verification reflect in the actual system test?

Reply:
It's a sanity check that we did not loose or duplicate any write.